cohttp-eio/src/client.ml

```diff
@@ -80,25 +104,56 @@ let tcp_address ~net uri =
   | ip :: _ -> ip
   | [] -> failwith "failed to resolve hostname"
 
+(* Create a socket for the uri, and signal whether it requires https *)
+let scheme_conn_of_uri ~sw net uri =
+  match Uri.scheme uri with
+  | Some "httpunix" ->
+      (* FIXME: while there is no standard, http+unix seems more widespread *)
+      `Plain (Eio.Net.connect ~sw net (unix_address uri) :> connection)
+  | Some "http" ->
+      `Plain (Eio.Net.connect ~sw net (tcp_address ~net uri) :> connection)
+  | Some "https" ->
+      `Https (Eio.Net.connect ~sw net (tcp_address ~net uri) :> connection)
```

Comment:
We move to this static signal via the `Https` variant, instead of applying the https wrap directly because, when using a proxy over http, we first need access to the socket flow unencrypted to set up the tunnel. But then we still need to wrap the socket in our tls encryption once the connection is established.

Reply:
Where is the HTTP -> HTTPS promotion happening? Is this in the pure proxy code?

Reply:
No it's ok.